deadlock-dotnet-sdk/CsWin32_NativeMethods.cs

```diff
@@ -0,0 +1,219 @@
+/// This file supplements code generated by CsWin32
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using Windows.Win32.System.Threading;
+using NTSTATUS_plus = PInvoke.NTSTATUS;
+using NTStatusException = PInvoke.NTStatusException;
+
+namespace Windows.Win32
+{
+    namespace Foundation
+    {
+        internal partial struct HANDLE : IComparable<HANDLE>
+        {
+            public static implicit operator HANDLE(nuint v) => new((nint)v);
+            public static implicit operator nuint(HANDLE v) => (nuint)(nint)v.Value;
+
+            public static implicit operator HANDLE(nint v) => new(v);
+            public static implicit operator nint(HANDLE v) => v.Value;
+
+            /// <summary>
+            /// Close the handle via the CloseHandle function
+            /// </summary>
+            /// <exception cref="Win32Exception">
+            /// If the application is running under a debugger, the function will throw an
+            /// exception if it receives either a handle value that is not valid or a
+            /// pseudo-handle value. This can happen if you close a handle twice, or if you
+            /// call CloseHandle on a handle returned by the FindFirstFile function instead
+            /// of calling the FindClose function.
+            /// </exception>
+            public void Close()
+            {
+                if (!PInvoke.CloseHandle(this))
+                    throw new Win32Exception();
```

Comment:
It might be a good idea to call this function here instead of throwing a generic Win32Exception:
https://learn.microsoft.com/en-us/windows/win32/api/errhandlingapi/nf-errhandlingapi-getlasterror

Reply:
This constructor calls the .NET wrapper for that function and gets the system message for the error code.
https://sourcegraph.com/github.com/dotnet/runtime@release/6.0/-/blob/src/libraries/Microsoft.Win32.Primitives/src/System/ComponentModel/Win32Exception.cs?L24:48

Reply:
https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.getlastpinvokeerror?view=net-6.0#remarks
GetLastPInvokeError reads a variable used only for P/Invoke win32 errors. This is different than GetLastWin32Error which the reads from a variable written also written to by the runtime.